recipes-kernel/kernel-tests/kernel-tests-files/kernel_dmesg_diff.py

```diff
@@ -106,9 +117,10 @@ def get_old_dmesg_log(db, logger):
     query = {}
     kernel_version = KernelVersion()
     os_version = OsVersion()
-    query['kernel_version_full'] = {'$lt': kernel_version.full}
+    query['kernel_version'] = { '$lt': kernel_version.version_dict }
```

Comment:
This works because of the order mongodb queries perform comparisons. It will actually go through the fields in order and compare each one to decide if the next one needs evaluated or not.